src/forwarder.ts

```diff
@@ -68,13 +75,35 @@ export async function addCloudWatchForwarderSubscriptions(
   const errors = [];
   if (typeof functionArn !== "string") {
     errors.push("Skipping forwarder ARN validation because forwarder string defined with CloudFormation function.");
-  } else if (integrationTesting === true) {
+  } else if (forwarderConfigs.IntegrationTesting === true) {
     errors.push("Skipping forwarder ARN validation because 'integrationTesting' is set to true");
   } else {
     await validateForwarderArn(aws, functionArn);
   }
   for (const [name, resource] of Object.entries(resources)) {
-    if (!isLogGroup(resource) || !resource.Properties.LogGroupName.startsWith("/aws/lambda/")) {
+    if (!isLogGroup(resource)) {
+      continue;
+    }
+    // if the extension is enabled, we don't want to subscribe to lambda log groups
+    if (
+      forwarderConfigs.AddExtension &&
+      !(
+        validateApiGatweaySubscription(resource, forwarderConfigs.SubToApiGatewayLogGroup) ||
+        validateHttpApiSubscription(resource, forwarderConfigs.SubToHttpApiLogGroup) ||
+        validateWebsocketSubscription(resource, forwarderConfigs.SubToWebsocketLogGroup)
+      )
+    ) {
+      continue;
+    }
+    // if the extension is disabled, we should subscribe to lambda log groups
+    if (
+      !(
+        resource.Properties.LogGroupName.startsWith("/aws/lambda/") ||
```

Comment:
When you find yourself writing complex branching conditions with lots of continues and breaks that boil down to a single decision, you can often simplify it to use a function with a bunch of return statements. Eg:
```typescript
function shouldSubscribe(resouce: any, forwarderConfigs: ForwarderConfigs): boolean {
   if (validateXXXXSubscription(resource, forwarderConfigs.SubToApiGatewayLogGroup) ){
      return true;
   }
   //... Other api gateway log groups types here etc etc
   if (!forwarderConfigs.AddExtension && resource.Properties.LogGroupName.startsWith("/aws/lambda/")) {
      return true;
   }
   return false;
}
```